R/g.analyse.perday.R

```diff
@@ -440,7 +472,14 @@ g.analyse.perday = function(ndays, firstmidnighti, time, nfeatures,
                 }
                 if (doquan == TRUE) {
                   q46 = c()
-                  q46 = quantile(varnum, probs = params_247[["qlevels"]], na.rm = T, type = quantiletype) * UnitReScale
+                  if (isFALSE(params_general[["WornDuringSleep"]])) {
```

Comment:
I did this only for the calculation of the percentiles, that's why I defined a new vector (`varnum_NA2zero`). The reason I did this is to keep consistency in the percentiles across days. For example, if recording started at 8am, the day 1 would have 16 hours of recording and the quantile 0.5 would represent the acceleration threshold over the **8** most active hours are accumulated (half of the recording). For the following days, the quantile 0.5 would correspond to the acceleration over the **12** most active hours are accumulated (half of the recording, for the next days would be 24 hours).

Reply:
> The reason I did this is to keep consistency in the percentiles across days. For example, if recording started at 8am, the day 1 would have 16 hours of recording and the quantile 0.5 would represent the acceleration threshold over the 8 most active hours are accumulated (half of the recording). For the following days, the quantile 0.5 would correspond to the acceleration over the 12 most active hours are accumulated (half of the recording, for the next days would be 24 hours).

Your approach assumes that the hours that are not recorded are all less active then the hours that are recorded. I am not sure this is justifiable assumption. The existing approach in the master branch assumes that the distribution for the hours we do record is a reasonable estimate for the distribution for a full 24 hours period. As a result the percentiles for the valid data act as an estimate for the percentiles for a 24 hours. For example M60 is still an estimate for M60 in a 24 hour day: Percentiles naturally account for differences in the duration of the underlying data, in contrast to time spent in intensity levels.

By default GGIR will exclude any day with less than 16 hours of valid data, which should address concerns that the first and last day are not representative. If the user is concerned about this then they can increase the includedaycrit threshold.

Reply:
Sounds good, I have reset this as in master, using averageday to impute the remaining part of the first and the last day (as it was). Only for the time-based metrics (ilevels, MVPA, recorded time), I have imputed NA values so that the sum of all ilevels categories does not go beyond the recorded time for the first and last day.